holmes/plugins/toolsets/opensearch/opensearch_logs.py

```diff
@@ -0,0 +1,185 @@
+                ),
+            },
+            toolset=toolset,
+        )
+        self._cache = None
+
+    def _invoke(self, params: Any) -> str:
+        err_msg = ""
+        try:
+            body = json.loads(params.get("query"))
+            full_query = body
+            full_query["size"] = int(
+                os.environ.get("OPENSEARCH_LOGS_SEARCH_SIZE", "5000")
+            )
+            logging.debug(f"opensearch logs search query: {full_query}")
+            headers = {"Content-Type": "application/json"}
+            headers.update(add_auth_header(self.toolset.config.opensearch_auth_header))
+
+            logs_response = requests.get(
+                url=self.toolset.config.opensearch_logs_url,
+                timeout=180,
+                verify=True,
+                data=json.dumps(full_query),
+                headers=headers,
+            )
+            if logs_response.status_code > 300:
+                err_msg = logs_response.text
+
+            logs_response.raise_for_status()
+            return json.dumps(logs_response.json())
+        except requests.Timeout:
+            logging.warn("Timeout while fetching opensearch logs search", exc_info=True)
+            return f"Request timed out while fetching opensearch logs search {err_msg}"
+        except RequestException as e:
+            logging.warn("Failed to fetch opensearch logs search", exc_info=True)
+            return f"Network error while opensearch logs search {err_msg} {str(e)}"
+        except Exception as e:
+            logging.warn("Failed to process opensearch logs search", exc_info=True)
+            return f"Unexpected error {err_msg}: {str(e)}"
+
+    def get_parameterized_one_liner(self, params) -> str:
+        return f'search logs: query="{params.get("query")}"'
+
+
+class OpenSearchLogsToolset(Toolset):
+    def __init__(self):
+        super().__init__(
+            name="opensearch/logs",
+            description="OpenSearch integration to fetch logs",
+            docs_url="",
+            icon_url="",
+            prerequisites=[CallablePrerequisite(callable=self.prerequisites_callable)],
+            tools=[
+                GetLogFields(toolset=self),
+                LogsSearchQuery(toolset=self),
+            ],
+            tags=[
+                ToolsetTag.CORE,
+            ],
+        )
+
+    def prerequisites_callable(self, config: dict[str, Any]) -> bool:
+        if not config and not os.environ.get("OPENSEARCH_LOGS_URL", None):
+            return False
+        elif not config and os.environ.get("OPENSEARCH_LOGS_URL", None):
+            self.config = OpenSearchLogsConfig(
+                opensearch_logs_url=os.environ.get("OPENSEARCH_LOGS_URL"),
+                opensearch_auth_header=os.environ.get(
+                    "OPENSEARCH_LOGS_AUTH_HEADER", None
+                ),
+            )
+            return True
+        else:
+            self.config = OpenSearchLogsConfig(**config)
+            return True
```

Comment:
I think the toolsets also need to implement `get_example_config` or the dynamic config in the UI will not be correct. e.g.

```
    def get_example_config(self):
        example_config = GrafanaLokiConfig(
            api_key="YOUR API KEY",
            url="YOUR GRAFANA URL",
            grafana_datasource_uid="<UID of the loki datasource to use>",
        )
        return example_config.model_dump()
```

Reply:
good point, fixed